src/contracts/interfaces/IPaymentCoordinator.sol

```diff
@@ -0,0 +1,175 @@
+import "./IPauserRegistry.sol";
+
+
+/**
+ * @title Interface for the `IPaymentCoordinator` contract.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @notice Allows AVSs to make "Range Payments", which get distributed amongst the AVSs' confirmed
+ * Operators and the Stakers delegated to those Operators.
+ * Calculations are performed based on the completed Range Payments, with the results posted in
+ * a Merkle root against which Stakers & Operators can make claims.
+ */
+interface IPaymentCoordinator {
+    /// STRUCTS ///
+    struct StrategyAndMultiplier {
+        IStrategy strategy;
+        // weight used to compare shares in multiple strategies against one another
+        uint96 multiplier;
+    }
+
+    struct RangePayment {
+        // Strategies & relative weights of shares in the strategies
+        StrategyAndMultiplier[] strategiesAndMultlipliers;
+        IERC20 token;
+        uint256 amount;
+        uint64 startTimestamp;
+        uint64 duration;
+    }
+
+    struct DistributionRoot {
+        // timestamp after which the root can be claimed against
+        uint64 activatedAfter;
+        // merkle root of the distribution
+        bytes32 root;
+    }
+
+    struct ClaimsTreeMerkleLeaf {
+        IERC20 token;
+        uint256 amount;
+        // The staker or operator who earned these tokens
+        address earner;
+    }
+
+    struct PaymentMerkleClaim {
+        ClaimsTreeMerkleLeaf[] leaves;
+        // The index of the root in the list of roots
+        uint32 rootIndex;
+        // proof of the claimaint's account root in the Merkle tree
+        uint32 accountIndex;
+        bytes accountTreeProof;
+        // The indices and proofs of the leafs in the claimaint's merkle tree for this root
+        uint32[] leafIndices;
+        bytes[] tokenTreeProofs;
+    }
+
+    /// EVENTS ///
+
+    event RangePaymentCreated(address indexed avs, bytes32 rangePaymentHash, RangePayment rangePayment);
+    event RangePaymentForAllCreated(address indexed submitter, bytes32 rangePaymentHash, RangePayment rangePayment);
+    event PaymentUpdaterSet(address indexed oldPaymentUpdater, address indexed newPaymentUpdater);
+    event PayAllForRangeSubmitterSet(address indexed payAllForRangeSubmitter, bool indexed oldValue, bool indexed newValue);
+    event ActivationDelaySet(uint64 oldActivationDelay, uint64 newActivationDelay);
+    event CalculationIntervalSecondsSet(uint64 oldCalculationIntervalSeconds, uint64 newCalculationIntervalSeconds);
+    event GlobalCommissionBipsSet(uint16 oldGlobalCommissionBips, uint16 newGlobalCommissionBips);
+    event RecipientSet(address indexed account, address indexed recipient);
+    event RootSubmitted(
+        bytes32 root,
+        uint32 paymentCalculationStartTimestamp,
+        uint32 paymentCalculationEndTimestamp,
+        uint32 activatedAt
+    );
+    event PaymentClaimed(ClaimsTreeMerkleLeaf leaf);
```

Comment:
I suppose the thought is these are "self" documenting? Might be good to @dev or @info these a bit, especially if they are emitted for multiple operations.

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/488/commits/4341741e9d7d6855c2b46ae67dca3970eb240f8d
Added some extra comments for some of these events.